core/src/main/java/org/apache/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -775,6 +779,10 @@ public Entity addNode(Location loc, Map<?,?> extraFlags) {
         entity.sensors().set(CLUSTER_MEMBER, true);
         entity.sensors().set(CLUSTER, this);
 
+        int nextClusterMemberId = getConfig(NEXT_CLUSTER_MEMBER_ID);
+        entity.sensors().set(CLUSTER_MEMBER_ID, nextClusterMemberId);
```

Comment:
Is this the wrong way round - should we set config on the new member to indicates its memberId, and have a sensor on the cluster that says what the next id should be? Or alternatively they could both be sensors (to be the same as `CLUSTER` and `CLUSTER_MEMBER` sensors already being set on the member)?

Is there a race if multiple members are being created concurrently? Perhaps we should extract out to a protected method something like `int obtainNextMemberId()`, which would do some synchronization and would thus increment the sensor atomically.